sympy/algebras/quaternion.py

```diff
@@ -304,6 +305,10 @@ def inverse(self):
         """Returns the inverse of the quaternion."""
         q = self
         return conjugate(q) * (1/q.norm()**2)
+        if not q.norm() :
+            raise ValueError("Cannot compute inverse of a quaternion with zero norm")
+        if q.norm() == oo :
```

Comment:
You should only focus on raising an error for infinite quaternion. Raising error for inverse of a zero quaternion is being fixed by PR #13693.